### ISSUE TITLE
Unexpected "Indent" in class body when the real problem is different, and in another line.

### ISSUE THREAD

**Issue by Ybaedbaenn** [godotengine/godot#79659]
### Godot version

v4.1.1.stable.official.bd6af8e0e

### System information

Windows 10.0.19045 - Vulkan (Compatibility) - GeForce GT 540M- Intel Core i7-2670QM CPU @ 2.20GHz (8 Threads)

### Issue description

Writing something after func definition, for example:

![image](https://github.com/godotengine/godot/assets/84321760/d361ccc1-4a6b-4880-9aaa-060f3cb6895c)

Causes an "Unexpected Indent in class body" error to appear one line below, as shown.

This message does not inform the user about the real problem in their code, and makes them look for a problem in a perfectly correct line below instead, confusing them and making debugging unnecessarily harder and longer.

### Steps to reproduce

Copy and paste this code, or write a similar one in the editor, then observe the bug:

```gdscript
extends Node


func _ready():K
	var rng = RandomNumberGenerator.new()


	rng.randomize()
	print(rng.seed)
```

### Minimal reproduction project

N/A

**Comment by dalexeev** [comment]
Yes, we have certain problems with cascading errors in the parser and analyzer. See also:

* #27702
* #75052

**Comment by marganec** [comment]
> Yes, we have certain problems with cascading errors in the parser and analyzer. See also:
> 
>     * [Single GDScript syntax mistake causes multiple errors #27702](https://github.com/godotengine/godot/issues/27702)
> 
>     * [Having .. gives incorrect error reports. #75052](https://github.com/godotengine/godot/issues/75052)

Uhh...? So, it wouldn't be fixed?

**Comment by dalexeev** [comment]
> So, it wouldn't be fixed?

No, I didn't mean it. Your issue is correct. I just mentioned probably related issues to make it easier to find them when someone will fix some one. :slightly_smiling_face:
